### PR TITLE
XWIKI-21209: Delete filter is slow (seems to load all filters)

### PR DESCRIPTION
**Pull request by surli**
JIRA: https://jira.xwiki.org/browse/XWIKI-21209

  * Refactor NotificationFilterPreferenceStore to ensure to use the right wiki in context when the local store is used
  * Use a query based on the filter internal identifier to perform single deletion, to prevent having to load it
  * Refactor the way NotificationFilterPreferenceDeletedEvent is used to not send the entire filter instance when it's triggered
  * Improve unit tests

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/internal/NotificationFilterPreferenceStore.java [hunk 1]
```diff
@@ -430,19 +436,21 @@ private void saveFilterPreferences(EntityReference entityReference,
...
+        } else if (wikiReference != null) {
+            context.setWikiReference(wikiReference);
         }
-
         try {
             return supplier.get();
         } finally {
-            if (!Objects.equals(currentWikiId, context.getWikiId())) {
-                context.setWikiId(currentWikiId);
+            if (wikiReference != null) {
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/internal/NotificationFilterPreferenceStore.java:452]: Not sure what is the point of this if, the context should get back to whatever was the context wiki and what's in this if is not related.

Comment (surli) [reply xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/internal/NotificationFilterPreferenceStore.java:452]: You're right, I don't know why I wrongly assumed that if there was no wikiReference parameter there was no point to reset it. It's only true with the localStore...

**Review by tmortagne — APPROVED** [review]
